src/Bitcoin/Address/Bech32.hs

```diff
@@ -1,5 +1,8 @@
+{-# LANGUAGE DuplicateRecordFields #-}
```

Comment:
I am on the fence about explicitly ambiguous record fields.  The case for allowing it here is pretty strong:
- The `*Result` types are unlikely to be used as type-class instances
- They are likely to be consumed quite near where they are produced

Some inconveniences:  
- In order to use, e.g. `result` for both types in the same module `DisambiguateRecordFields` is needed (correct me here if I'm wrong)
- Downstream code _might_ require more explicit type annotations as a result

Reply:
Yup, `DisambiguateRecordFields` can be used if you want to use the record field but have both record fields imported in your module with the same, or no,  qualifier. Since GHC 9.20, you can also now also just turn on `OverloadedRecordDot` and use `x.result` anywhere without any real downsides. I tend to turn on `DuplicateRecordFields` very liberally, but it would be nice if record fields had their own namespace and weren't automatically functions with the same without some explicit intent.

How do ambiguous record fields interact with type class instances that make them more inconvenient? Just curious.

Reply:
Yeah it would be great if you could disambiguate by using `Ctor.someField` or something.

Hm.  I guess it's just an intuitive concern that ambiguous record fields can hurt type inference.  Sometimes code is accidentally more general than it seems.  Suppose you have a type class where `Bech32EncodeResult` is a instance but `Bech32DecodeResult` is not.  If you wrote some code (unannotated like in a `where` clause) which was accidentally polymorphic over these two types, it would not be possible to compose it with the type class members without providing an additional annotation.  To contrast, if the fields are unambiguous the code might end up being monotype instead.

Reply:
I see. I've pushed the changes with the field names disambiguated. Added it for all fields to keep it consistent as they may share more fields later on.